smc.s

```diff
@@ -0,0 +1,634 @@
+#
+
+# We check 3 types of modification of ALU instructions:
+# * Modification of immediate (addi %r1, %r2, $5 -> addi %r1, %r2, $a5)
+# * Modification of register (addi %r1, %r2, $5 -> addi %r3, %r1, $5)
+# * Modification of opcode (add %r1, %r2, %r2 -> sub %r1, %r2, %r2)
+#
+# After that, we check modification of unconditional jump targets
+#
+# Each check begins with the simplest case:
+# modifying instruction is executed ~10 instructions
+# before instruction it modifies, than the gap becomes
+# closer: 4, 3, 2, 1, 0 filling instructions.
+# We use different gaps to check behavior
+# on different pipeline depths.
+#
+# The SMC subroutine template goes as follows
+#   fill_TYPE_GAPLEN:
+#       la $t1, NEW_LABEL   # get address of new instruction
+#       lw $t0, 0($t1)      # read new instruction
+#       la $t1, OLD_LABEL   # get address of overwritten instruction
+#       sw $t0, 0($t1)      # overwrite instruction
+#     # gap of GAPLEN instruction goes here
+#   OLD_LABEL:
+#       add $t1, $t2, 0x5
+#     # Checking subroutine. If it fails, we jump to `fail' label
+#     # to stop simulation
+
+.globl __start
+
+.data
+fm:	.asciiz "Failed test\n"
+
+.text
+
+
+__start: 
+
+    j main
+
+#instructions that we want to get after modification
+new1:
+    add $t1, $t2, 0xa5
+new2:
+    add $t3, $t1, 0x5
+new3:
+    sub $t1, $t2, $t2
+#print fail message
+fail:	
+    li $v0, 4	# syscall 4 (print_str)
+	la $a0, fm
+	syscall
+    jr $ra
+	#li $v0, 10	# syscall 10 (exit)
+    #syscall
+
+main:
+
+#--------------------------------------------
+# check SMC for modification if immediate
+#--------------------------------------------
+
+#--------------------------------------------
+
+    
+#gap  with 10 instructions
+fill1_10:
+
+    
+    la $t1, new1    # t1 = &(new_instruction)
+    lw $t0, 0($t1)  # t0 = new_instruction
+    la $t1, smc1_10 # t1 = &(old_instruction)
+    
+    sw $t0, 0($t1)  # *(old_instruction) = new_insrtuction
+
+    #gap 
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+
+    #old instruction   
+smc1_10:
+    add $t1, $t2, 0x5
+
+    #check modification. if wasn't  modificated then print fail message  
+    la $s2, smc1_10 # s2 = &(modificated_instruction)
+    lw $s2, 0($s2)  # s2 = modificated_instruction
+    beq $t0, $s2, fill1_4 # if new_insrtuction == modificated_instruction 
+                          # then continue test with another gaps
+                          # else print fail message
+    jal fail
+    #continue testing another gap
+#-----------------------------------------------
+
+#gap with 4 instructions
+fill1_4:
+
+    #modificate
+    la $t1, smc1_4 
+    sw $t0, 0($t1)
+
+    #gap
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+    add $s0, $s1, $s2
+
+    #old instruction
+smc1_4:
```

Comment:
Same thing here. Labels have to be clear, we have enough space in our repository for additional letters.